spec/30-processing-model.md

```diff
@@ -0,0 +1,66 @@
+
+## Processing Model in Forwarding Mode
+
+In forwarding mode a tracing system decides to forward tracing information but to not modify the trace-context headers. The request is then considered as not sampled. 
+
+1. Tracing system receives a `traceparent` header and may receive `tracecontext`. 
+2. Tracing system tries to parse the version of the `traceparent` header.
+   1. If the version prefix cannot be parsed (it's not 2 hex characters followed by
+   dash (`-`)), implementation should restart the trace.
+*NOTE* : If a system is in forwarding only mode. It cannot start a trace. 
+   2. If the version is higher than supported by the tracing system, does no further validations. 
+3. When the version number is supported by the tracing system it validates `trace-id` and `parent-id`
+  1. If `trace-id` and `parent-id` are valid, the tracing system adds the entire `trace-context` to all outgoing request orginating from the current incoming request. 
+  2. If either `trace-id` or `parent-id` are invalid, the tracing systems does not add the trace context to any outgoing requests.
+*NOTE* : We never explictly defined this, but I think this is what should happen. 
+
+## Processing Model in Participation Mode
+
+In participation mode the tracing system decides to forward AND modofiy trace-context headers. The request is then consideres as sampled.
+
+1. Tracing system receives a `traceparent` header and may receive `tracecontext`. 
+2. Tracing system tries to parse the version of the `traceparent` header.
+  1. If the version prefix cannot be parsed (it's not 2 hex characters followed by
+   dash (`-`)), the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`. 
+  2. If the version number is higher than supported by the tracing implemenation, the implementation uses the format defined in this specification to parse `trace-id` and `parent-id`. The tracing system will only parse `flags` values supported by the current version of  this specification and ignore all other values. If parsing fails, the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`. 
+3. When the version number is supported by the tracing system it validates `trace-id` and `parent-id`.
+  1. If either `trace-id`, `parent-id` or `flags`  are invalid, the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`.
+4. After the system has validate the traceparent it validates the tracestate
+*NOTE* : How do we do this exactly?
+
+5. Tracing system modifies the `traceparent` header. 
+1. **Update `parent-id`**. The value of property `parent-id` can be set to the
```

Comment:
so you copied section on mutation here and haven't removed the source. Is it intentional?